### PR TITLE
Clarifying use of flag

### LINKED ISSUE #15790: `export DOCKER_CONTENT_TRUST=1` versus `sudo docker`
Description of problem:

For those invoking `docker` with sudo (rather than running as root or adding their user to the docker group) it's likely that the DOCKER_CONTENT_TRUST env variable won't be passed when using a default sudo setup.

`docker version`:

```
Client:
 Version:      1.8.1
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   d12ea79
 Built:        Thu Aug 13 02:32:18 UTC 2015
 OS/Arch:      linux/amd64

Server:
 Version:      1.8.1
 API version:  1.20
 Go version:   go1.4.2
 Git commit:   d12ea79
 Built:        Thu Aug 13 02:32:18 UTC 2015
 OS/Arch:      linux/amd64
```

`docker info`:

```
Containers: 738
Images: 1639
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 3116
 Dirperm1 Supported: false
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.13.0-29-generic
Operating System: Ubuntu precise (12.04.5 LTS)
CPUs: 1
Total Memory: 993.7 MiB
Name: docker
ID: OEQ3:4OFN:Q4WS:EW5C:HIFM:RDCV:Y6CE:ZE3I:OC3R:FDUB:24FN:XRXV
Username: cpswan
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
```

`uname -a`:

    Linux docker 3.13.0-29-generic #53~precise1-Ubuntu SMP Wed Jun 4 22:06:25 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux

Environment details (AWS, VirtualBox, physical, etc.): VMware 5.5 VM

How reproducible:

```
$ export DOCKER_CONTENT_TRUST=1
$ sudo docker pull cpswan/hello_onug
```

Steps to Reproduce:
1. set the env variable `DOCKER_CONTENT_TRUST=1`
2. invoke `sudo docker` for a command that requires trust

Actual Results:

```
Using default tag: latest
Pulling repository docker.io/cpswan/hello_onug
183f0a0d3b62: Download complete
d3cc6fa2aa29: Download complete
9217ba43cf44: Download complete
339c97963d06: Download complete
Status: Image is up to date for cpswan/hello_onug:latest
```

Expected Results:

```
Using default tag: latest
no trust data available
```

Additional info:

The following workarounds may be used:

    sudo -E docker ...

Or 'sudo visudo' and add the following line:

    Defaults        env_keep += "DOCKER_CONTENT_TRUST"

### PR DESCRIPTION
**Pull request by moxiegirl**
Closes #15790 

Signed-off-by: Mary Anthony mary@docker.com

### CODE REVIEW

File: docs/security/trust/content_trust.md [hunk 1]
```diff
@@ -152,10 +152,21 @@ To enable content trust in a `bash` shell enter the following command:
 export DOCKER_CONTENT_TRUST=1
 ```
 
-Once set, each of the "tag" operations require key for trusted tag. All of these
-commands also support the `--disable-content-trust` flag. This flag allows
-publishers to run individual operations on tagged images without content trust on an
-as-needed basis. 
+Once set, each of the "tag" operations require key for trusted tag.
```
Comment (thaJeztah) [inline docs/security/trust/content_trust.md]: s/require key/require a key/

Comment (thaJeztah) [inline docs/security/trust/content_trust.md]: Hm, for *a* trusted tag?

TL;DR I think there's something missing in the sentence, LOL

File: docs/security/trust/content_trust.md [hunk 2]
```diff
@@ -152,10 +152,21 @@ To enable content trust in a `bash` shell enter the following command:
 export DOCKER_CONTENT_TRUST=1
 ```
 
-Once set, each of the "tag" operations require key for trusted tag. All of these
-commands also support the `--disable-content-trust` flag. This flag allows
-publishers to run individual operations on tagged images without content trust on an
-as-needed basis. 
+Once set, each of the "tag" operations require key for trusted tag. 
+
+In an environment where `DOCKER_CONTENT_TRUST` is set, you can use the `--disable-content-trust` flag to run individual operations on tagged images without content trust on an as-needed basis.
```
Comment (thaJeztah) [inline docs/security/trust/content_trust.md]: Just a nit; could you wrap to 80-chars (this line and the line below)

**Comment by thaJeztah** [comment]
Couple of nits, LGTM otherwise.

(Reading this; If `--disable-content-trust=false` becomes a frequently used feature, I think we should add a dedicated `--enable-content-trust` flag)

**Comment by moxiegirl** [comment]
Done.  @cpswan @endophage

File: docs/security/trust/content_trust.md [hunk 3]
```diff
@@ -152,23 +152,38 @@ To enable content trust in a `bash` shell enter the following command:
...
+
+In an environment where `DOCKER_CONTENT_TRUST` is set, you can use the
+`--disable-content-trust` flag to run individual operations on tagged images
+without content trust on an as-needed basis.
+
+```bash
+$  docker pull --disable-content-trust docker/trusttest:untrusted
+```
+
+You can also use this flag in a shell where you haven't enabled trust to turn a
```
Comment (endophage) [inline docs/security/trust/content_trust.md]: Can simplify this a little. This command will enable content trust regardless of whether the ENV variable is set or not.

File: docs/security/trust/content_trust.md [hunk 4]
```diff
@@ -177,31 +192,32 @@ choose a long, complex passphrase and be careful to keep the password and the
...
- 
-The passphrase you chose for both the offline key and your content key-pair should
-be randomly generated and stored in a *password manager*. 
 
-It is important to note, if you had left off the `latest` tag, content trust is skipped.
-This is true even if content trust is enabled and even if this is your first push. 
+The passphrase you chose for both the offline key and your content key-pair
+should be randomly generated and stored in a *password manager*.
+
+It is important to note, if you had left off the `latest` tag, content trust is
```
Comment (endophage) [inline docs/security/trust/content_trust.md]: Not something that will be fixed right now, but it would be awesome to have some kind of standardized callout box to highlight things like this in the docs.

**Comment by endophage** [comment]
I'd make the one small simplification, otherwise LGTM.

File: docs/security/trust/content_trust.md (hunk 3 above)
Comment (cpswan) [inline docs/security/trust/content_trust.md]: I'm with @endophage here. The docs simply need to state that --disable-content-trust=false will invoke a command with content trust enabled (regardless of how the ENV variable is set).

File: docs/security/trust/content_trust.md (hunk 4 above)
Comment (moxiegirl) [inline docs/security/trust/content_trust.md]: I can use a note for this.   I'll add that in.

File: docs/security/trust/content_trust.md (hunk 3 above)
Comment (moxiegirl) [inline docs/security/trust/content_trust.md]: I'll update with that clarification.

**Comment by moxiegirl** [comment]
ping @thaJeztah for the merge

File: docs/security/trust/content_trust.md [hunk 5]
```diff
@@ -177,31 +190,31 @@ choose a long, complex passphrase and be careful to keep the password and the
...
-The passphrase you chose for both the offline key and your content key-pair should
-be randomly generated and stored in a *password manager*. 
 
-It is important to note, if you had left off the `latest` tag, content trust is skipped.
-This is true even if content trust is enabled and even if this is your first push. 
+The passphrase you chose for both the offline key and your content key-pair
+should be randomly generated and stored in a *password manager*.
+
+> **NOTE**: If you omit the `latest` tag, content trust is skipped. This is true
+even if content trust is enabled and even if this is your first push.
```
Comment (thaJeztah) [inline docs/security/trust/content_trust.md]: nit; the second line of the note should also start with `>` (not worth fixing)

**Comment by thaJeztah** [comment]
LGTM, thanks!
